File: backend/models/student.model.js
/**
 * Student document schema (Firestore collection: 'students')
 *
 * @typedef {Object} Student
 * @property {string} id - Firestore document ID
 * @property {string} name - Full name
 * @property {string} email - Email address
 * @property {string} rollNumber - School roll number
 * @property {string} class - Class/Grade (e.g., "Class V")
 * @property {string} section - Section (e.g., "A")
 * @property {string} dateOfBirth - ISO date string
 * @property {string} guardianName - Parent/guardian name
 * @property {string} guardianContact - Parent/guardian phone
 * @property {string} address - Home address
 * @property {'active'|'inactive'|'graduated'} status - Enrollment status
 * @property {string} createdAt - ISO timestamp
 * @property {string} updatedAt - ISO timestamp
 */

export function validateStudent(data) {
  const required = ['name', 'rollNumber', 'class'];
  const missing = required.filter((field) => !data[field]);
  if (missing.length > 0) {
    throw new Error(`Missing required fields: ${missing.join(', ')}`);
  }
  return true;
}

File: backend/models/student_model.py
"""
Student document schema (Firestore collection: 'students')

Fields:
  id            - Firestore document ID
  name          - Full name
  email         - Email address
  rollNumber    - School roll number
  class         - Class/Grade (e.g., "Class V")
  section       - Section (e.g., "A")
  dateOfBirth   - ISO date string
  guardianName  - Parent/guardian name
  guardianContact - Parent/guardian phone
  address       - Home address
  status        - Enrollment status: 'active' | 'inactive' | 'graduated'
  createdAt     - ISO timestamp
  updatedAt     - ISO timestamp
"""


def validate_student(data):
    required = ['name', 'rollNumber', 'class']
    missing = [field for field in required if not data.get(field)]
    if missing:
        raise ValueError(f"Missing required fields: {', '.join(missing)}")
    return True
